zerver/migrations/0507_fix_emoji_metadata.py

```diff
@@ -0,0 +1,98 @@
+import re
+from typing import Any, Dict
+
+import boto3
+from botocore.client import Config
+from django.conf import settings
+from django.db import migrations
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+from django.db.models import Exists, OuterRef
+
+
+def fix_emoji_metadata(apps: StateApps, schema_editor: BaseDatabaseSchemaEditor) -> None:
+    if settings.LOCAL_UPLOADS_DIR is not None:
+        return
+
+    bucket = boto3.resource(
+        "s3",
+        aws_access_key_id=settings.S3_KEY,
+        aws_secret_access_key=settings.S3_SECRET_KEY,
+        region_name=settings.S3_REGION,
+        endpoint_url=settings.S3_ENDPOINT_URL,
+        config=Config(
+            signature_version=None,
+            s3={"addressing_style": settings.S3_ADDRESSING_STYLE},
+        ),
+    ).Bucket(settings.S3_AVATAR_BUCKET)
+
+    Realm = apps.get_model("zerver", "Realm")
+    RealmEmoji = apps.get_model("zerver", "RealmEmoji")
+    UserProfile = apps.get_model("zerver", "UserProfile")
+
+    for realm in Realm.objects.filter(
+        Exists(RealmEmoji.objects.filter(realm_id=OuterRef("id"), deactivated=False))
+    ).only("id"):
+        fallback_owner = (
+            UserProfile.objects.filter(realm=realm.id, is_bot=False, role=100, is_active=True)
+            .order_by("id")
+            .first()
+        )
+        assert fallback_owner is not None
+
+        emoji_by_filename: Dict[str, Any] = {}
+        for emoji in RealmEmoji.objects.filter(realm_id=realm.id, deactivated=False).exclude(
+            file_name=None
+        ):
```

Comment:
I think filtering for `deactivated=False` might be wrong, if we want the invariant to be true for all custom emoji and there's a possible future where we allow reactivating them? I think they're included in exports even if deactivated, right?

Reply:
Yeah, we do include them in exports even if deactivated.